pages/_app.js

```diff
@@ -186,6 +190,48 @@ export default function App({
   function handleResetFilter() {
     setSearchTerm("");
   }
+  useEffect(() => {
+    if ("serviceWorker" in navigator && "PushManager" in window) {
+      window.addEventListener("beforeinstallprompt", (event) => {
+        event.preventDefault();
+        setDeferredPrompt(event);
+        setShowInstallButton(true);
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener("beforeinstallprompt", (event) => {
+      event.preventDefault();
+      setDeferredPrompt(event);
+      setShowInstallButton(true);
+      setShowInstallPrompt(false);
+    });
+
+    window.addEventListener("appinstalled", () => {
+      setShowInstallPrompt(false);
+      setShowInstallButton(true);
+      console.log("App installed");
+      setDeferredPrompt(null);
+    });
+
+    return () => {
+      window.removeEventListener("beforeinstallprompt", () => {});
+      window.removeEventListener("appinstalled", () => {});
```

Comment:
this does not remove the eventListeners. You need to store the callback function in a variable and reference it in the removeEventListener call:
```js
function prepareInstallation() { ...}
window.addEventListener("beforeinstallprompt", prepareInstallation)
return () => {
        window.removeEventListener("beforeinstallprompt", prepareInstallation);
}
```